plugins/database/postgresql/postgresql.go

```diff
@@ -188,6 +226,15 @@ func (p *PostgreSQL) changeUserPassword(ctx context.Context, username string, ch
 				"username": username,
 				"password": password,
 			}
+
+			if p.PasswordEncryption == PasswordEncryptionSCRAMSHA256 {
+				hashedPassword, err := scram.Encrypt(password)
+				if err != nil {
+					return fmt.Errorf("unable to scram-sha256 password: %w", err)
+				}
+				m["password"] = hashedPassword
+			}
```

Comment:
Can we freely switch from plaintext to encrypted when generating new passwords for existing users? Might not be an issue but I'm wondering if, for example, this works:
* config plugin with scram encryption
* create new user with hashed password
* update config to plaintext encryption
* try to update the user password
Does it work?

If yes then all good, if not or there are corner cases we might need to prevent users from changing the encryption strategy of existing configs.

Reply:
yes, the current password_encryption from the config will be respected, so it's possible to change the encryption strategy even after it's already been set.